File: webApp/src/utils/validators.js
/** Validation and formatting helpers. */

export function isValidEmail(email) {
  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
}

export function isValidPassword(password) {
  return typeof password === "string" && password.length >= 6
}

/** Validates a Brazilian CPF using the official check-digit algorithm. */
export function isValidCPF(cpf) {
  const cleaned = String(cpf).replace(/\D/g, "")
  if (cleaned.length !== 11) return false
  if (/^(\d)\1{10}$/.test(cleaned)) return false

  const calcDigit = (slice) => {
    let sum = 0
    for (let i = 0; i < slice.length; i++) {
      sum += Number(slice[i]) * (slice.length + 1 - i)
    }
    const rest = (sum * 10) % 11
    return rest === 10 ? 0 : rest
  }

  const digit1 = calcDigit(cleaned.slice(0, 9))
  const digit2 = calcDigit(cleaned.slice(0, 10))
  return digit1 === Number(cleaned[9]) && digit2 === Number(cleaned[10])
}

export function formatCPF(value) {
  return String(value)
    .replace(/\D/g, "")
    .slice(0, 11)
    .replace(/(\d{3})(\d)/, "$1.$2")
    .replace(/(\d{3})(\d)/, "$1.$2")
    .replace(/(\d{3})(\d{1,2})$/, "$1-$2")
}

export function formatPrice(value) {
  return new Intl.NumberFormat("pt-BR", { style: "currency", currency: "BRL" }).format(value)
}

export function translateAuthError(message = "") {
  const map = {
    "Invalid login credentials": "E-mail ou senha incorretos.",
    "User already registered": "Este e-mail já está cadastrado.",
    "Email not confirmed": "Confirme seu e-mail antes de entrar.",
    "Password should be at least 6 characters": "A senha deve ter ao menos 6 caracteres.",
  }
  return map[message] || message || "Ocorreu um erro. Tente novamente."
}

File: mobile_app/lib/core/themes/utils/validators.dart
class Validators {
  static bool isValidEmail(String email) {
    final emailRegex = RegExp(
      r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$',
    );
    return emailRegex.hasMatch(email);
  }

  static bool isValidCPF(String cpf) {
    String cleaned = cpf.replaceAll(RegExp(r'[^\d]'), '');

    if (cleaned.length != 11) return false;
    if (RegExp(r'^(\d)\1{10}$').hasMatch(cleaned)) return false;

    int sum = 0;
    for (int i = 0; i < 9; i++) {
      sum += int.parse(cleaned[i]) * (10 - i);
    }
    int firstDigit = 11 - (sum % 11);
    if (firstDigit >= 10) firstDigit = 0;

    if (int.parse(cleaned[9]) != firstDigit) return false;

    sum = 0;
    for (int i = 0; i < 10; i++) {
      sum += int.parse(cleaned[i]) * (11 - i);
    }
    int secondDigit = 11 - (sum % 11);
    if (secondDigit >= 10) secondDigit = 0;

    return int.parse(cleaned[10]) == secondDigit;
  }

  static String formatCPF(String cpf) {
    String cleaned = cpf.replaceAll(RegExp(r'[^\d]'), '');
    if (cleaned.length <= 11) {
      if (cleaned.length > 9) {
        return '${cleaned.substring(0, 3)}.${cleaned.substring(3, 6)}.${cleaned.substring(6, 9)}-${cleaned.substring(9)}';
      } else if (cleaned.length > 6) {
        return '${cleaned.substring(0, 3)}.${cleaned.substring(3, 6)}.${cleaned.substring(6)}';
      } else if (cleaned.length > 3) {
        return '${cleaned.substring(0, 3)}.${cleaned.substring(3)}';
      }
    }
    return cleaned;
  }

  static bool isValidPassword(String password) {
    return password.length >= 6;
  }
}
